Include/pyhash.h

```diff
@@ -12,8 +12,21 @@ PyAPI_FUNC(Py_hash_t) _Py_HashPointer(void*);
 PyAPI_FUNC(Py_hash_t) _Py_HashBytes(const void*, Py_ssize_t);
 #endif
 
-/* Prime multiplier used in string and various other hashes. */
-#define _PyHASH_MULTIPLIER 1000003UL  /* 0xf4243 */
+/* Constant used for various hashes */
+#define _PyHASH_MULTIPLIER ((Py_uhash_t)1000003UL)
+
+/* Official constants from the xxHash specification */
+#if SIZEOF_VOID_P > 4
```

Comment:
Despite at least one confused comment in the current code, the size of a hash code has no logical relationship to the size of a pointer.  We're doing arithmetic on ints of type `Py_uhash_t`, so the obvious thing to test instead is `SIZEOF_PY_UHASH_T`.

Reply:
Sure, I didn't know that `SIZEOF_PY_UHASH_T` existed.

Reply:
By the way, I used `SIZEOF_VOID_P` because I copied that from here:
```
#if SIZEOF_VOID_P >= 8
#  define _PyHASH_BITS 61
#else
#  define _PyHASH_BITS 31
#endif
```

Should I also change that to use `SIZEOF_PY_UHASH_T`?